ui/src/core_components/base/BaseMarkdown.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+	<div
+		v-if="isMarkedLoaded"
+		v-dompurify-html="marked.parse(props.rawText).trim()"
+		class="BaseMarkdown markdown"
+	></div>
+</template>
+
+<script setup lang="ts">
+import { onBeforeMount, ref } from "vue";
+
+let marked: typeof import("marked");
+const isMarkedLoaded = ref(false);
+
+const props = defineProps<{
+	rawText: string;
+}>();
+
+onBeforeMount(async () => {
+	marked = await import("marked");
+	isMarkedLoaded.value = true;
```

Comment:
Yes, it's needed, I looked into this as I was wondering the same thing.

Turns out it's executed before the DOM is mounted, but it doesn't await the function that you pass. So if mounting the DOM takes 5ms and loading `marked` takes 100ms (supposing it's not cached), then the variable `marked` will be empty at the time of render.

I could wrap `marked` in a `ref`, specifically a `shallowRef` to avoid the overhead, but I thought it was more descriptive to create `isMarkedLoaded`.